src/sphinxcontrib/towncrier/ext.py

```diff
@@ -123,7 +124,7 @@ def _nodes_from_document_markup_source(
     node.document = state.document
     nested_parse_with_titles(
         state=state,
-        content=statemachine.ViewList(
+        content=statemachine.StringList(
```

Comment:
This seems to be more correct, per https://github.com/sphinx-doc/sphinx/pull/12836. Hope it doesn't backfire.